src/server/channel_store.cc

```diff
@@ -23,6 +23,30 @@ bool Matches(string_view pattern, string_view channel) {
   return stringmatchlen(pattern.data(), pattern.size(), channel.data(), channel.size(), 0) == 1;
 }
 
+// Build functor for sending messages to connection
+auto BuildSender(string_view channel, facade::ArgRange messages) {
+  size_t messages_size = accumulate(messages.begin(), messages.end(), 0,
+                                    [](int sum, string_view str) { return sum + str.size(); });
+  auto buf = shared_ptr<char[]>{new char[channel.size() + messages_size]};
+  {
+    memcpy(buf.get(), channel.data(), channel.size());
+    char* ptr = buf.get() + channel.size();
+    for (string_view message : messages) {
+      memcpy(ptr, message.data(), message.size());
+      ptr += message.size();
+    }
+  }
+
+  return [channel, buf = std::move(buf), messages](facade::Connection* conn, string pattern) {
```

Comment:
to stress further my previous point, messages refer to a memory that can be destroyed by the time this callback runs, no?

Reply:
Ah, yes, good catch 😵